### PR TITLE
[PL-49112] Update SMP K8s supported versions

### PR DESCRIPTION
**Pull request by brian-f-harness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Update SMP K8s supported versions
* Jira/GitHub Issue numbers (if any): PL-49112
* Preview links/images (Internal contributors only): [Supported Kubernetes versions](https://66214e5eddf022125f430e6f--harness-developer.netlify.app/docs/self-managed-enterprise-edition/smp-supported-platforms#supported-kubernetes-versions)

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by brian-f-harness** [comment]
@raghuAtWings @gn-harness @bharatgoelharness @pratmit 

Do we want to keep this bullet? 
Harness commits to support new minor versions of Kubernetes within three months of the first stable release. For example, if the stable release occurs on August 31, Harness extends compatibility by November 30.

File: docs/self-managed-enterprise-edition/shared/smp-supported-platforms.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-Solution| Supported Platform| Connected | HA | Monitoring| Disaster Recovery | 
-| :-----------------------------------------------------------------: | :----------------------------: | :-----------: | :----------------: | :-------------------: | :-----------------: | 
+Solution| Supported Platform| Connected | HA | Monitoring| Disaster Recovery |
+| :-----------------------------------------------------------------: | :----------------------------: | :-----------: | :----------------: | :-------------------: | :-----------------: |
 | Kubernetes Cluster| Kubernetes - GKE - AKS - EKS | Yes| Yes | Prometheus, Grafana | Yes
 
 ### Supported Kubernetes versions
 
-* Self-Managed Enterprise Edition supports Kubernetes v.1.27, as well as versions 1.26, 1.25, 1.24, 1.23, 1.22, 1.21, and 1.20.
+* Self-Managed Enterprise Edition supports Kubernetes v.1.28, as well as versions 1.27, 1.26, 1.25, 1.24, 1.23, 1.22, 1.21, and 1.20.
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/shared/smp-supported-platforms.md:9]:
```suggestion
* Self-Managed Enterprise Edition supports Kubernetes v.1.28 and 1.27.
```

**Review by pratmit — COMMENTED** [review]
Made change on proposing why we support only these versions.

File: docs/self-managed-enterprise-edition/shared/smp-supported-platforms.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-Solution| Supported Platform| Connected | HA | Monitoring| Disaster Recovery | 
-| :-----------------------------------------------------------------: | :----------------------------: | :-----------: | :----------------: | :-------------------: | :-----------------: | 
+Solution| Supported Platform| Connected | HA | Monitoring| Disaster Recovery |
+| :-----------------------------------------------------------------: | :----------------------------: | :-----------: | :----------------: | :-------------------: | :-----------------: |
 | Kubernetes Cluster| Kubernetes - GKE - AKS - EKS | Yes| Yes | Prometheus, Grafana | Yes
 
 ### Supported Kubernetes versions
 
-* Self-Managed Enterprise Edition supports Kubernetes v.1.27, as well as versions 1.26, 1.25, 1.24, 1.23, 1.22, 1.21, and 1.20.
+* Self-Managed Enterprise Edition supports Kubernetes v.1.28 and 1.27. We intend to support the active k8s versions as k8s versions are released.
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/shared/smp-supported-platforms.md:9]:
```suggestion
* Self-Managed Enterprise Edition supports Kubernetes v.1.28 and 1.27. We intend to support active K8s versions as they are released. 
```

**Pull request merged**